payments-core/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -51,18 +60,100 @@ internal abstract class BaseAddPaymentMethodFragment(
         val viewBinding = FragmentPaymentsheetAddPaymentMethodBinding.bind(view)
         addPaymentMethodHeader = viewBinding.addPaymentMethodHeader
 
-        addPaymentMethodFragment()
+        val config = requireNotNull(
+            arguments?.getParcelable<FragmentConfig>(
+                BaseSheetActivity.EXTRA_FRAGMENT_CONFIG
```

Comment:
ooc what were the tradeoffs on sending in the config vs just the paymentMethodTypes?  I wonder if there is value in just passing the paymentMethod types which may be specified from the paymentIntent, but also possible some sort of pre-filter and special ordering.

This class currently is awesome in that you were able to keep any logic functions really out of it.   This is the only thing that has some logic like converting between PaymentMethodTypes and SupportedPaymentMethod and also for if the resulting list is empty.  I wonder if it would be simplified by passing in a known valid list of payment methods?  Not sure

Reply:
Moved the check to the ViewModel when the StripeIntent is set. If there's no intersection between the paymentMethods in the intent and the ones Payment Sheet supports, fail early.